aws-cpp-sdk-core/source/http/curl/CurlHandleContainer.cpp

```diff
@@ -205,12 +215,6 @@ bool CurlHandleContainer::CheckAndGrowPool()
 
             if (curlHandle)
             {
-                //for timeouts to work in a multi-threaded context,
-                //always turn signals off. This also forces dns queries to
-                //not be included in the timeout calculations.
-                curl_easy_setopt(curlHandle, CURLOPT_NOSIGNAL, 1L);
-                curl_easy_setopt(curlHandle, CURLOPT_TIMEOUT_MS, m_requestTimeout);
-                curl_easy_setopt(curlHandle, CURLOPT_CONNECTTIMEOUT_MS, m_connectTimeout);
```

Comment:
No, because `CheckAndGrowPool()` is only called from `AcquireCurlHandle()`, so after the former has allocated a new handle, the latter will always apply the options.